### PR TITLE
Add tldr for openssl

### PR DESCRIPTION
**Pull request by Duologic**
An excerpt from https://www.sslshopper.com/article-most-common-openssl-commands.html

### CODE REVIEW

**Comment by ostera** [comment]
Hi and thanks for contributing @Duologic, could you please use braced names instead of numbers or urls? Think `{{url}}` instead o `www.paypal.com`.

Same goes for files, `{{path/to/certificate}}` rather than `certificate.crt`, to maintain consisten with the rest of the pages.

Other than that this looks good to be merged :+1: 

Thanks again!

**Comment by igorshubovych** [comment]
I agree with @ostera.

Also please use one command code per description.

If necessary, you can add more examples.

**Comment by Duologic** [comment]
Sure, no problem here it is.

File: pages/common/openssl.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+`openssl s_client -connect {{url}}:443`
+
+- Generate new private key and CSR
+
+`openssl req -out {{path/to/CSR.csr}} -new -newkey rsa:2048 -nodes -keyout {{path/to/pivate.key}}`
+
+- Read contents of certificate, private key and CSR file
+
+`openssl x509 -text -noout -in {{path/to/certificate.crt}}`
+`openssl rsa -check -in {{path/to/pivate.key}}`
```
Comment (igorshubovych) [inline pages/common/openssl.md]: This should be  a separate example

File: pages/common/openssl.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+`openssl req -text -noout -verify -in {{path/to/CSR.csr}}`
```
Comment (igorshubovych) [inline pages/common/openssl.md]: This should be a separate example

File: pages/common/openssl.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+- Check if certificate, private key and CSR have the same MD5 hash
+
+`openssl x509 -noout -modulus -in {{path/to/certificate.crt}} | openssl md5`
+`openssl rsa -noout -modulus -in {{path/to/pivate.key}} | openssl md5`
```
Comment (igorshubovych) [inline pages/common/openssl.md]: This should be a separate example

File: pages/common/openssl.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+`openssl req -noout -modulus -in {{path/to/CSR.csr}} | openssl md5`
```
Comment (igorshubovych) [inline pages/common/openssl.md]: This should be a separate example

File: pages/common/openssl.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# openssl
+
+> OpenSSL is a cryptography toolkit.
+
+- Check an SSL connection
+
+`openssl s_client -connect {{url}}:443`
```
Comment (sbrl) [inline pages/common/openssl.md]: Suggestion: Change `{{url}}` to `{{domain}}`. A url would be `https://example.com/testing?foo=bar....`, but I think that openssl takes something more like simply `example.com`.

**Comment by Duologic** [comment]
There we go, one command per description + {{domain}}:{{port}} for the SSL connection command.

**Comment by igorshubovych** [comment]
@Duologic now page becomes a bit bloated. 8 examples versus recommended 5-6. I would agree with 7.
From your practice, what are the least used command here.

**Comment by Duologic** [comment]
Aargh, pick something out.

**Comment by sbrl** [comment]
Perhaps we should do something similar to the `git` family of pages? Maybe we could have a main master page with a few of the most common ones, and a 'see also' section detailing the subcommands and how to reach them (like `openssl-enc` for example).

**Comment by Duologic** [comment]
Semantically you could split the openssl commands up in openssl-req, openssl-rsa and openssl-x509 and a general openssl. But my experience with openssl is that I use the commands in this file mostly in tandem with each other. Placing these commands in different files wouldn't be as useful as putting them together.

**Comment by Duologic** [comment]
Anyone who can make a decision on where to go with this pull request?

**Comment by waldyrious** [comment]
@igorshubovych @ostera can you comment?

**Comment by ostera** [comment]
@Duologic I think it's fine for now, if we run into more examples that are specific to different openssl sub executables we can refactor. Squash and I'll merge if @igorshubovych is onboard

**Comment by igorshubovych** [comment]
LGTM. But plz squash the commits.

**Comment by rubenvereecken** [comment]
There was a new PR waiting for openssl at #682 by @notpeter. @Duologic could you finish up?

**Comment by igorshubovych** [comment]
@Duologic thank you for you for contribution! Keep up the good work!

**Comment by Duologic** [comment]
Sorry I didn't squash the commits, didn't have enough free time.

I really like the project, keep up the good work!

**Comment by igorshubovych** [comment]
NP, we recently decided we will not be that harsh with commits to simplify contribution

**Pull request merged**